tests/test_crm.py

```diff
@@ -0,0 +1,487 @@
+
+    # get neighbor and arp entry
+    dvs.servers[0].runcmd("ping -c 1 10.0.0.0")
+    dvs.servers[1].runcmd("ping -c 1 10.0.0.2")
+
+    db = swsscommon.DBConnector(0, dvs.redis_sock, 0)
+    ps = swsscommon.ProducerStateTable(db, "ROUTE_TABLE")
+    fvs = swsscommon.FieldValuePairs([("nexthop","10.0.0.1,10.0.0.3"), ("ifname", "Ethernet0,Ethernet4")])
+
+    time.sleep(5*60)
+
+    # get counters
+    used_counter = getCrmCounterValue(dvs, 'crm_stats_nexthop_group_object_used')
+    avail_counter = getCrmCounterValue(dvs, 'crm_stats_nexthop_group_object_available')
+
+    # add route and update available counter
+    ps.set("2.2.2.0/24", fvs)
+    setReadOnlyAttr(dvs, 'SAI_OBJECT_TYPE_SWITCH', 'SAI_SWITCH_ATTR_AVAILABLE_NEXT_HOP_GROUP_ENTRY', '999')
+
+    time.sleep(60)
+
+    # get counters
+    new_used_counter = getCrmCounterValue(dvs, 'crm_stats_nexthop_group_object_used')
+    new_avail_counter = getCrmCounterValue(dvs, 'crm_stats_nexthop_group_object_available')
+
+    assert new_used_counter - used_counter == 1
+    assert avail_counter - new_avail_counter == 1
+
+    # remove route and update available counter
+    ps._del("2.2.2.0/24")
+    setReadOnlyAttr(dvs, 'SAI_OBJECT_TYPE_SWITCH', 'SAI_SWITCH_ATTR_AVAILABLE_NEXT_HOP_GROUP_ENTRY', '1000')
+
+    time.sleep(60)
+
+    # get counters
+    new_used_counter = getCrmCounterValue(dvs, 'crm_stats_nexthop_group_object_used')
+    new_avail_counter = getCrmCounterValue(dvs, 'crm_stats_nexthop_group_object_available')
+
+    assert new_used_counter == used_counter
+    assert new_avail_counter == avail_counter
+
+
+def test_CrmNexthopGroupMember(dvs):
+
+    dvs.runcmd("ifconfig Ethernet0 10.0.0.0/31 up")
+    dvs.runcmd("ifconfig Ethernet4 10.0.0.2/31 up")
+
+    dvs.servers[0].runcmd("ifconfig eth0 10.0.0.1/31")
+    dvs.servers[0].runcmd("ip route add default via 10.0.0.0")
+
+    dvs.servers[1].runcmd("ifconfig eth0 10.0.0.3/31")
+    dvs.servers[1].runcmd("ip route add default via 10.0.0.2")
+
+    dvs.runcmd("crm config polling interval 1")
+
+    setReadOnlyAttr(dvs, 'SAI_OBJECT_TYPE_SWITCH', 'SAI_SWITCH_ATTR_AVAILABLE_NEXT_HOP_GROUP_MEMBER_ENTRY', '1000')
+
+    # get neighbor and arp entry
+    dvs.servers[0].runcmd("ping -c 1 10.0.0.0")
+    dvs.servers[1].runcmd("ping -c 1 10.0.0.2")
+
+    db = swsscommon.DBConnector(0, dvs.redis_sock, 0)
+    ps = swsscommon.ProducerStateTable(db, "ROUTE_TABLE")
+    fvs = swsscommon.FieldValuePairs([("nexthop","10.0.0.1,10.0.0.3"), ("ifname", "Ethernet0,Ethernet4")])
+
+    time.sleep(5*60)
+
+    # get counters
+    used_counter = getCrmCounterValue(dvs, 'crm_stats_nexthop_group_member_used')
+    avail_counter = getCrmCounterValue(dvs, 'crm_stats_nexthop_group_member_available')
+
+    # add route and update available counter
+    ps.set("2.2.2.0/24", fvs)
+    setReadOnlyAttr(dvs, 'SAI_OBJECT_TYPE_SWITCH', 'SAI_SWITCH_ATTR_AVAILABLE_NEXT_HOP_GROUP_MEMBER_ENTRY', '998')
```

Comment:
Since we increment CRM_NEXTHOP_GROUP_MEMBER for every group member add, irrespective of the group object, just wondering what will be the SAI return value for SAI_..AVAILABLE_NEXT_HOP_GROUP_MEMBER_ENTRY

Reply:
SAI also returns global number of available members regardless of the group object.